daml-lf/spec/daml-lf-1.rst

```diff
@@ -3094,41 +3094,93 @@ as described by the ledger model::
 
      'tpl' (x : T) ↦ { 'precondition' eₚ, … }  ∈  〚Ξ〛Mod
      eₚ[x ↦ vₜ]  ⇓  Err E
-   —————————————————————————————————————————————————————————————————————— EvUpdCreateErr1
+   —————————————————————————————————————————————————————————————————————— EvUpdCreateErr1a
      'create' @Mod:T vₜ ‖ S₀  ⇓ᵤ  (Err E, ε)
 
      'tpl' (x : T) ↦ { 'precondition' eₚ, … }  ∈  〚Ξ〛Mod
      eₚ[x ↦ vₜ]  ⇓  Ok 'False'
-   —————————————————————————————————————————————————————————————————————— EvUpdCreateFail
+   —————————————————————————————————————————————————————————————————————— EvUpdCreateFail1a
      'create' @Mod:T vₜ ‖ S₀
        ⇓ᵤ
      (Err (Fatal "Precondition failed on {Mod:T}."), ε)
 
-     'tpl' (x : T) ↦ { 'precondition' eₚ, 'agreement' eₐ, … }  ∈  〚Ξ〛Mod
+     'tpl' (x : T) ↦ { 'precondition' eₚ, …,
+        'implements' Mod₁:I₁ { … }, …, 'implements' Modₖ:Iₖ { … } }  ∈  〚Ξ〛Mod
      eₚ[x ↦ vₜ]  ⇓  Ok 'True'
+     'interface' (x₁ : I₁) ↦ { …, 'precondition' e₁ₚ, … } ∈ 〚Ξ〛Mod₁
+     e₁ₚ[x₁ ↦ 'to_interface' @Mod₁:I₁ @Mod:T vₜ]  ⇓  Ok 'True'
+     'interface' (x₂ : I₂) ↦ { …, 'precondition' e₂ₚ, … } ∈ 〚Ξ〛Mod₂
+     e₂ₚ[x₂ ↦ 'to_interface' @Mod₂:I₂ @Mod:T vₜ]  ⇓  Ok 'True'
+        …
+     'interface' (xᵢ : Iᵢ) ↦ { …, 'precondition' eᵢₚ, … } ∈ 〚Ξ〛Modᵢ
+     eᵢₚ[x₂ ↦ 'to_interface' @Modᵢ:Iᵢ @Mod:T vₜ]  ⇓  Err E
+   —————————————————————————————————————————————————————————————————————— EvUpdCreateErr1b
+     'create' @Mod:T vₜ ‖ S₀  ⇓ᵤ  (Err E, ε)
+
+     'tpl' (x : T) ↦ { 'precondition' eₚ, …,
+        'implements' Mod₁:I₁ { … }, …, 'implements' Modₖ:Iₖ { … } }  ∈  〚Ξ〛Mod
+     eₚ[x ↦ vₜ]  ⇓  Ok 'True'
+     'interface' (x₁ : I₁) ↦ { …, 'precondition' e₁ₚ, … } ∈ 〚Ξ〛Mod₁
+     e₁ₚ[x₁ ↦ 'to_interface' @Mod₁:I₁ @Mod:T vₜ]  ⇓  Ok 'True'
+     'interface' (x₂ : I₂) ↦ { …, 'precondition' e₂ₚ, … } ∈ 〚Ξ〛Mod₂
+     e₂ₚ[x₂ ↦ 'to_interface' @Mod₂:I₂ @Mod:T vₜ]  ⇓  Ok 'True'
+        …
+     'interface' (xᵢ : Iᵢ) ↦ { …, 'precondition' eᵢₚ, … } ∈ 〚Ξ〛Modᵢ
+     eᵢₚ[xᵢ ↦ 'to_interface' @Modᵢ:Iᵢ @Mod:T vₜ]  ⇓  Ok 'False'
```

Comment:
```suggestion
     'interface' (xₖ : Iₖ) ↦ { …, 'precondition' eₖₚ, … } ∈ 〚Ξ〛Modₖ
     eₖₚ[xₖ ↦ 'to_interface' @Modₖ:Iₖ @Mod:T vₜ]  ⇓  Ok 'False'
```

Reply:
I think we do want `xᵢ` here since the last implemented interface is not necessarily the one that failed

Reply:
This going together with the previous change:
``'implements' Modₖ:Iₖ { … }, …``, i.e. `xₖ` is not necessarily the last one. 

It is just a suggestion anyway.

Reply:
ah I see what you meant now. I now think it's better as it is, without either change, since it matches the definition, where `ImplDefₖ` _is_ the last `'implements'` sub-definition and also the last sub-definition overall.

```
       |  'tpl' (x : T) ↦                           -- DefTemplate
            { 'precondition' e₁
            , 'signatories' e₂
            , 'observers' e₃
            , 'agreement' e₄
            , 'choices' { ChDef₁, …, ChDefₘ }
            , KeyDef
            , ImplDef₁, …, ImplDefₖ
            }
```